libraries/autoplayDetection/autoplay.js

```diff
@@ -33,8 +33,13 @@ function startDetection() {
   videoElement.setAttribute('playsinline', 'true');
   videoElement.muted = true;
 
-  videoElement
-    .play()
+  const videoPlay = videoElement.play();
+  if (!videoPlay) {
+    autoplayEnabled = false;
+    return;
+  }
+
+  videoPlay
```

Comment:
I think this could just be `Promise.resolve(videoElement.play())`.

Could you add a test case for the regression - if .play() doesn't return a promise, it shouldn't throw?

Reply:
Should `autoplayEnabled` be `true` when `.play()` does not return a promise?